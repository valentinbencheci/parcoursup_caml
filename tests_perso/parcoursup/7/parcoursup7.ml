(* ###################### - DESCRIPTION TEST - ###################### *)
(* On verifie comment fonctionne l'algo quand un candidat a une grande 
liste d'attente et puis, tous les formations lui sont disponibles. 
En plus, on verifie ce que se passe quand Emma refuse une formation 
qu'elle n'a pas choisi.*)
(* ################################################################## *)

open Test_parcoursup_utils

let session = nouvelle_session ()

let () =
  ajoute_candidat session ~nom_candidat:"Adam";
  ajoute_candidat session ~nom_candidat:"Leo";
  ajoute_candidat session ~nom_candidat:"Raphael";
  ajoute_candidat session ~nom_candidat:"Louise";
  ajoute_candidat session ~nom_candidat:"Alice";
  ajoute_candidat session ~nom_candidat:"Emma";
  ajoute_formation session ~nom_formation:"Université Côte d'Azur" ~capacite:1;
  ajoute_formation session ~nom_formation:"Université de Toulon" ~capacite:1;
  ajoute_formation session ~nom_formation:"Université de Lyon" ~capacite:1;
  ajoute_formation session ~nom_formation:"Université de Paris" ~capacite:1;
  ajoute_formation session ~nom_formation:"Université de Grenoble" ~capacite:1;

  ajoute_voeu session
    ~rang_repondeur:(Some 1)
    ~nom_candidat:"Adam"
    ~nom_formation:"Université Côte d'Azur";
  ajoute_voeu session
    ~rang_repondeur:(Some 1)
    ~nom_candidat:"Leo"
    ~nom_formation:"Université de Toulon";
  ajoute_voeu session
    ~rang_repondeur:(Some 1)
    ~nom_candidat:"Raphael"
    ~nom_formation:"Université de Lyon";
  ajoute_voeu session
    ~rang_repondeur:(Some 1)
    ~nom_candidat:"Louise"
    ~nom_formation:"Université de Paris";
  ajoute_voeu session
    ~rang_repondeur:(Some 1)
    ~nom_candidat:"Alice"
    ~nom_formation:"Université de Grenoble";

  ajoute_voeu session
    ~rang_repondeur:(Some 5)
    ~nom_candidat:"Emma"
    ~nom_formation:"Université Côte d'Azur";
  ajoute_voeu session
    ~rang_repondeur:(None)
    ~nom_candidat:"Emma"
    ~nom_formation:"Université de Toulon";
  ajoute_voeu session
    ~rang_repondeur:(Some 0)
    ~nom_candidat:"Emma"
    ~nom_formation:"Université de Lyon";
  ajoute_voeu session
    ~rang_repondeur:(Some 0)
    ~nom_candidat:"Emma"
    ~nom_formation:"Université de Paris";
  ajoute_voeu session
    ~rang_repondeur:(None)
    ~nom_candidat:"Emma"
    ~nom_formation:"Université de Grenoble";

  ajoute_commission session
    ~nom_formation:"Université Côte d'Azur"
    ["Adam";"Leo";"Emma";"Louise";"Alice";"Raphael"];
  ajoute_commission session
    ~nom_formation:"Université de Toulon"
    ["Adam";"Leo";"Emma";"Louise";"Alice";"Raphael"];
  ajoute_commission session
    ~nom_formation:"Université de Lyon"
    ["Adam";"Leo";"Emma";"Louise";"Alice";"Raphael"];
  ajoute_commission session
    ~nom_formation:"Université de Paris"
    ["Adam";"Leo";"Emma";"Louise";"Alice";"Raphael"];
  ajoute_commission session
    ~nom_formation:"Université de Grenoble"
    ["Adam";"Leo";"Emma";"Louise";"Alice";"Raphael"];

  reunit_commissions session;

  nouveau_jour session;
  affiche_voeux_en_attente session "Adam";
  affiche_propositions_en_attente session "Adam";
  affiche_voeux_en_attente session "Leo";
  affiche_propositions_en_attente session "Leo";
  affiche_voeux_en_attente session "Raphael";
  affiche_propositions_en_attente session "Raphael";
  affiche_voeux_en_attente session "Louise";
  affiche_propositions_en_attente session "Louise";
  affiche_voeux_en_attente session "Alice";
  affiche_propositions_en_attente session "Alice";
  affiche_voeux_en_attente session "Emma";
  affiche_propositions_en_attente session "Emma";

  renonce session
    ~nom_candidat:"Adam"
    ~nom_formation:"Université Côte d'Azur";
    renonce session
    ~nom_candidat:"Leo"
    ~nom_formation:"Université de Toulon";
    renonce session
    ~nom_candidat:"Raphael"
    ~nom_formation:"Université de Lyon";
    renonce session
    ~nom_candidat:"Louise"
    ~nom_formation:"Université de Paris";
  renonce session
    ~nom_candidat:"Alice"
    ~nom_formation:"Université de Grenoble";

  nouveau_jour session;
  affiche_voeux_en_attente session "Adam";
  affiche_propositions_en_attente session "Adam";
  affiche_voeux_en_attente session "Leo";
  affiche_propositions_en_attente session "Leo";
  affiche_voeux_en_attente session "Raphael";
  affiche_propositions_en_attente session "Raphael";
  affiche_voeux_en_attente session "Louise";
  affiche_propositions_en_attente session "Louise";
  affiche_voeux_en_attente session "Alice";
  affiche_propositions_en_attente session "Alice";
  affiche_voeux_en_attente session "Emma";
  affiche_propositions_en_attente session "Emma";

  renonce session
    ~nom_candidat:"Emma"
    ~nom_formation:"Université de Paris";
  renonce session
    ~nom_candidat:"Emma"
    ~nom_formation:"Université de Lyon";
  renonce session
    ~nom_candidat:"Emma"
    ~nom_formation:"Université de Toulon";
  renonce session
    ~nom_candidat:"Emma"
    ~nom_formation:"Université de Grenoble";

  nouveau_jour session;
  affiche_voeux_en_attente session "Adam";
  affiche_propositions_en_attente session "Adam";
  affiche_voeux_en_attente session "Leo";
  affiche_propositions_en_attente session "Leo";
  affiche_voeux_en_attente session "Raphael";
  affiche_propositions_en_attente session "Raphael";
  affiche_voeux_en_attente session "Louise";
  affiche_propositions_en_attente session "Louise";
  affiche_voeux_en_attente session "Alice";
  affiche_propositions_en_attente session "Alice";
  affiche_voeux_en_attente session "Emma";
  affiche_propositions_en_attente session "Emma";