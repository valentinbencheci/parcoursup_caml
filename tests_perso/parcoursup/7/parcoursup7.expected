nouvelle session
ajoute candidat Adam
ajoute candidat Leo
ajoute candidat Raphael
ajoute candidat Louise
ajoute candidat Alice
ajoute candidat Emma
ajoute formation Université Côte d'Azur (1 places)
ajoute formation Université de Toulon (1 places)
ajoute formation Université de Lyon (1 places)
ajoute formation Université de Paris (1 places)
ajoute formation Université de Grenoble (1 places)
ajoute voeu de Adam pour Université Côte d'Azur (rang=1)
ajoute voeu de Leo pour Université de Toulon (rang=1)
ajoute voeu de Raphael pour Université de Lyon (rang=1)
ajoute voeu de Louise pour Université de Paris (rang=1)
ajoute voeu de Alice pour Université de Grenoble (rang=1)
ajoute voeu de Emma pour Université Côte d'Azur (rang=5)
ajoute voeu de Emma pour Université de Toulon
ajoute voeu de Emma pour Université de Lyon (rang=0)
ajoute voeu de Emma pour Université de Paris (rang=0)
ajoute voeu de Emma pour Université de Grenoble
commission de classement de Université Côte d'Azur : [Adam; Leo; Emma; Louise; Alice; Raphael]
commission de classement de Université de Toulon : [Adam; Leo; Emma; Louise; Alice; Raphael]
commission de classement de Université de Lyon : [Adam; Leo; Emma; Louise; Alice; Raphael]
commission de classement de Université de Paris : [Adam; Leo; Emma; Louise; Alice; Raphael]
commission de classement de Université de Grenoble : [Adam; Leo; Emma; Louise; Alice; Raphael]
réunit commissions
nouveau jour
Voeux en attente de Adam : {}
Propositions en attente de Adam : {Université Côte d'Azur}
Voeux en attente de Leo : {}
Propositions en attente de Leo : {Université de Toulon}
Voeux en attente de Raphael : {Université de Lyon(0)}
Propositions en attente de Raphael : {}
Voeux en attente de Louise : {Université de Paris(0)}
Propositions en attente de Louise : {}
Voeux en attente de Alice : {Université de Grenoble(0)}
Propositions en attente de Alice : {}
Voeux en attente de Emma : {Université Côte d'Azur(0), Université de Toulon(0)}
Propositions en attente de Emma : {Université de Lyon, Université de Paris, Université de Grenoble}
Adam renonce à Université Côte d'Azur
Leo renonce à Université de Toulon
Raphael renonce à Université de Lyon
Louise renonce à Université de Paris
Alice renonce à Université de Grenoble
nouveau jour
Voeux en attente de Adam : {}
Propositions en attente de Adam : {}
Voeux en attente de Leo : {}
Propositions en attente de Leo : {}
Voeux en attente de Raphael : {}
Propositions en attente de Raphael : {}
Voeux en attente de Louise : {}
Propositions en attente de Louise : {}
Voeux en attente de Alice : {}
Propositions en attente de Alice : {}
Voeux en attente de Emma : {}
Propositions en attente de Emma : {Université de Toulon, Université de Grenoble, Université de Paris, Université de Lyon}
Emma renonce à Université de Paris
Emma renonce à Université de Lyon
Emma renonce à Université de Toulon
Emma renonce à Université de Grenoble
nouveau jour
Voeux en attente de Adam : {}
Propositions en attente de Adam : {}
Voeux en attente de Leo : {}
Propositions en attente de Leo : {}
Voeux en attente de Raphael : {}
Propositions en attente de Raphael : {}
Voeux en attente de Louise : {}
Propositions en attente de Louise : {}
Voeux en attente de Alice : {}
Propositions en attente de Alice : {}
Voeux en attente de Emma : {}
Propositions en attente de Emma : {}
